### ISSUE TITLE
User not permitted to access library "Media Folders"

### ISSUE THREAD

**Issue by WisdomCode** [jellyfin/jellyfin#1436]
As already  mentioned in #1347, I have problems with permissions. These are unfortunately relatively numerous, in short it is impossible to use the webpage thats referring to a piece of media to access it. 
For example:
The actors page loads no videos, leaving this area empty
The same for artists when using a music library
Also, I cannot access the singular titles of an album through the album page
The same goes when going to a genres page, for example, all adventure movies. The page remains empty, as there were no entries at all.

The combination of these issues make also searching nearly impossible, as you have to search the media itself, not the album, actor or genre, as clicking on them will only present the empty page. Empty is in this case not referring to no metadata about the actor or artist, but simply no associated media shown.

All these errors have the same effect on the jellyfloss logs:
"USERNAME" is not permitted to access Library "Media Folders".

This also happens to the admin user, when it is not allowed to access all libraries (choosing accessible libraries by hand). As to reproducing, it has been this way since I installed it (pretty positive it was version 3.6.1), no importing or similar quirks were performed, it was a simple and clean installation with 2 users, an admin and a 'consumer'.

I am using Debian 9 (amd64), it's installed locally via apt.

**Comment by pjeanjean** [comment]
I tried to debug this, and from what I could understand all the 'virtuals' folders (i.e. actors, artists, genres and everything else that exists in `Media Folders`) are defined as `UserRootFolder` items.
A quick fix is thus to disable permissions checking added with #930 for items of this type, but I don't know if it could break something else as I can't really find any documentation for the items types...

**Comment by WisdomCode** [comment]
For now, my workaround which is working pretty well (without recompiling etc.) is simply giving the user access to all libraries. This fixes the issue. For restricting the access to a certain library, I give the specific library a rating of XXX (or something similar) while forbidding the user to access to content rated that high. This may not work for all scenarios, but for many, as there are many ratings available. 
Hopefully, this is useful to someone with similar problems. Audio playback is fixed too this way (see #1347 ). 
Additionally I hope that I won't need this workaround for too long. Thank you for investigating @pjeanjean  and thanks in advance to all other developers trying to resolve this!

**Comment by CTalvio** [comment]
Collections are still broken on a user without full library access. If a user searches an item specifically, it can now be viewed and played, unlike before 10.3.5. But items are still not displayed when viewing collections that contain them.

![vivaldi_2019-06-10_23-38-13](https://user-images.githubusercontent.com/4365015/59225241-e0aad680-8bd8-11e9-8065-84dd825c67b1.png)
![vivaldi_2019-06-10_23-38-25](https://user-images.githubusercontent.com/4365015/59225243-e0aad680-8bd8-11e9-9193-7e3f13cec6d2.png)

**Comment by pjeanjean** [comment]
I am still unsure of how collections are managed actually...
You will need #1394 as well, but even with these two fixes I keep having strange behaviours with collections (and I don't have the time to dig deeper in the codebase right now :/ )

**Comment by DrPandemic** [comment]
This doesn't seem fixed.

**Comment by pjeanjean** [comment]
The particular issue with actors, artists and genres should be fixed.
The one about albums is a duplicate of #1347.

**Comment by DrPandemic** [comment]
Thanks @pjeanjean for letting me know. I'll try to clean this issue and https://github.com/jellyfin/jellyfin/issues/1436 once https://github.com/jellyfin/jellyfin/pull/1394/ is merged.

**Comment by DrPandemic** [comment]
https://github.com/jellyfin/jellyfin/pull/1394 was merged and seems to be fixing the issues.

**Comment by Aanok** [comment]
This does not seem fixed for good.

When a user's Access permissions do not "Enable access to all libraries" but selectively allow only some (or all) of them on an individual basis, the top Folder view is presented empty regardless of permissions.

This is an example configuration for a test case:
![Screenshot_2020-03-26 Jellyfin](https://user-images.githubusercontent.com/18417628/77637644-442ea680-6f56-11ea-9867-fd243002587a.png)
![Screenshot_2020-03-26 Jellyfin(1)](https://user-images.githubusercontent.com/18417628/77637650-46910080-6f56-11ea-9b90-3f17cf1c76a4.png)

This is the response you get for an example query run with the above test user:
```
curl -i -H "x-emby-token: ${token}" "${server}/users/${userid}/items?parentid=${foldersviewid}"

HTTP/2 200 
server: nginx/1.14.2
date: Thu, 26 Mar 2020 10:42:32 GMT
content-type: application/json; charset=utf-8
content-length: 48
expires: 0
access-control-allow-headers: Accept, Accept-Language, Authorization, Cache-Control, Content-Disposition, Content-Encoding, Content-Language, Content-Length, Content-MD5, Content-Range, Content-Type, Date, Host, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since, Origin, OriginToken, Pragma, Range, Slug, Transfer-Encoding, Want-Digest, X-MediaBrowser-Token, X-Emby-Authorization
access-control-allow-methods: GET, POST, PUT, DELETE, PATCH, OPTIONS
access-control-allow-origin: *
strict-transport-security: max-age=63072000

{"Items":[],"TotalRecordCount":0,"StartIndex":0}
```

Of note: **Jellyfin correctly honours access permissions for individual items** (e.g. a TV show, a song or even a whole library), it is just the top Folders view that is inaccessible no matter what.

Of further note: trying to access a forbidden resource returns an empty list of items like above and an HTTP status code of 200. There should really be indication of what's happening with either a 403 status code or similar, or an error in the JSON itself.


**System:**
 - OS: Debian 9.12
 - Virtualization: None, installed from apt
 - Clients: All
 - Browser: Firefox 74
 - Jellyfin Version: 10.5.2
 - Installed Plugins: AudioDB 10.5.2.0, MusicBrainz 10.5.2.0, Open Subtitles 7.0.0.0
 - Reverse Proxy: nginx, but also on direct connection
 - Base URL: /jellyfin
 - Networking: Host
 - Storage: Local

**To Reproduce**
1. Configure a user with selective access to libraries, making sure the Folders view is enabled.
2. Try and access the Folders view.

**Expected behavior**
The user can access the Folders view.

**Logs**
```
[2020-03-26 11:43:02.893 +01:00] [WRN] "test_user" is not permitted to access Library "Folders".
```

**Comment by JustAMan** [comment]
@Aanok I suggest opening a new issue linking to this one, as this one has quite a length of discussion which isn't necessarily relevant to the bug.

**Comment by Aanok** [comment]
> @Aanok I suggest opening a new issue linking to this one, as this one has quite a length of discussion which isn't necessarily relevant to the bug.

Righto, done :)
